CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+- Add a custom parameter to avoid a debugging problem.  To use the compiler, go to _File_ → _Settings_ → _Compiler_ → _Java Compiler_ → _Additional command line parameters_ and add `-Xep:ParameterName:OFF`
+- Tests: please follow the conventions of the Maven Surefire plugin for [unit tests](https://maven.apache.org/surefire/maven-surefire-plugin/examples/inclusion-exclusion.html) and those of the Maven Failsafe plugin for [integration tests](https://maven.apache.org/surefire/maven-failsafe-plugin/examples/inclusion-exclusion.html). To run the integration tests just run the verify phase, e.g.: `mvn clean verify`
+
+### Understanding branching
+
+We follow a [GitFlow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) approach to our project.
+This can be summarized as:
+1. `master` represents what is running in production, or is in the process of being released for a new deployment
+2. `gbif-dev` represents the latest state of development, and is what is run by the [continuous build tools](https://builds.gbif.org/)
+   4. Generally feature branches are made from here
+3. `ala-dev` represents a large feature branch to bring in the `livingatlas` module and pipelines for the ALA work
+
+If you are working on a new feature and not part of the core team please ask for guidance on where to start (most likely `gbif-dev`).
+
+### Make your change
+
+1. Checkout the branch you need (or fork the project and then checkout the branch)
+2. Create a feature branch following a naming convention of `<issue_number>_my_new_feature`
+3. Add unit tests for your change (please see testing style below)
+3. Arrange your commits
+    1. Consider merging commits. We favour fewer commits, but recognize this is not always desirable for large features (please squash small commits addressing typos etc. into one)
+4. Use descriptive commit messages that make it easy to identify changes and provide a clear history.
+    1. Please reference the issue you are working on unless it is a trivial change
+    2. Examples of good commit messages:
+          1. `#123 Enable occurrenceStatus interpretation to avro`
+          1. `Fixup: Addressing typos in JDoc`
+    2. Examples of bad commit messages:
+          1. ` `
+          1. `Various fixes`
+          1. `#123`
+5. Check your code compiles, and all project unit tests pass (please be kind to reviewers)
+6. Explore the `errorprone` warnings raised at compilation time. Please address issues you see as best you can.
+7. Ensure that the code is spotless (`mvn spotless:check` and fixup styling issues using `mvn spotless:apply`)
+8. Verify that the PR only changes the code necessary to address the issue (other fixes should be in separate PRs)
+6. Prefer to create a pull request and have it reviewed for larger submissions. Committers are free to push *smaller* changes directly.
+
```

Comment:
Should we add "Delete the branch after PR"? I am not sure that a user will be able to do that without some permissions

Reply:
Perhaps the person with permission to merge should do that.